src/controller/CHIPDeviceController.cpp

```diff
@@ -706,16 +709,16 @@ CHIP_ERROR DeviceCommissioner::EstablishPASEConnection(NodeId remoteDeviceId, Re
     {
         if (params.HasConnectionObject())
         {
-            SuccessOrExit(err = mSystemState->BleLayer()->NewBleConnectionByObject(params.GetConnectionObject()));
+            SuccessOrExit(err = mSystemState->BleLayer()->NewBleConnectionByObject(params.GetConnectionObject()), PASESessionBLE);
         }
         else if (params.HasDiscoveredObject())
         {
             // The RendezvousParameters argument needs to be recovered if the search succeed, so save them
             // for later.
             mRendezvousParametersForDeviceDiscoveredOverBle = params;
-            SuccessOrExit(err = mSystemState->BleLayer()->NewBleConnectionByObject(params.GetDiscoveredObject(), this,
-                                                                                   OnDiscoveredDeviceOverBleSuccess,
-                                                                                   OnDiscoveredDeviceOverBleError));
+            SuccessOrExit(err = mSystemState->BleLayer()->NewBleConnectionByObject(
+                              params.GetDiscoveredObject(), this, OnDiscoveredDeviceOverBleSuccess, OnDiscoveredDeviceOverBleError),
+                          PASESessionBLE);
```

Comment:
@anush-apple ?

Reply:
Yes, so this was not clear at all with the extension done for the macro. I maintain that `SuccessOrExit` and other CodeUtils.h macros that are pervasively used should not be overridden to have particular tracing semantics under the same name. We could easily add `SuccessOrExitWithTracing` or such in tracing.h, that has a dependency on SuccessOrExit macro. That way we don't change the existing macro in ways that are not obviously tracing.

I am making the request on readability grounds, given that I review a lot of this code, and I already find the pervasive use of `SuccessOrExit` adds much mental burden to reviews in general (there is a hidden goto you always have to think about), since many ALSO have side effects.

This SuccessOrExit is really really tricky:

- Have to scan a really long line.
- 1st side effect -> `err = XXXX` lvalue set
- 2nd side effect -> Perhaps a `goto`, often really far away (as here, 20+ lines down)
- 3rd side effect (added now) -> a logging of a metric key, only on failure, that looks like a class name (see my comment about keeping the `kMetricPaseBleSession` constant tracking, which is also easier to ctrl-click in IDES. 

It may seem "Oh it's not that tricky", but given that this is a critical complex method, with many many paths, and compile-time #ifdefs, and it's a 135 lines method with multiple exit paths, some with clean-up, some not, it really adds up.

At the very least:

`SuccessOrExitWithTracing(kMetricPaseBleSession, err = xxxxx)` would:

- Only 1 more macro of 1 line defined
- Be clear to watch-out for a metrics side effect (and you can SEARCH for all of the places a metric is added on failure by searching `SuccessOrExitWithTracing`, rather than having to sift through all SuccessOrExit (writing a regex to find the ones with > 1 args is a literal adventure, given `,` can appear nested in the arguments.
- Make it easier to find metrics with the `kMetric[A-Za-z0-9]+` regex for common metrics, which would catch there an more, with a well-defined kMetric regex...

Reply:
I believe this is resolved